build/yaml/botbuilder-dotnet-ci-slacktest.yml

```diff
@@ -0,0 +1,86 @@
+trigger:
+  - master
+
+pr:
+  - master
+
+pool:
+  vmImage: 'windows-2019'
+
+variables:
+  BotGroup: 'TestSlackBot-Group-CI-Build'
+  BotName: 'TestSlackBot-Echo-CI-Build'
+  BuildConfiguration: 'debug'
+  BuildPlatform: 'any cpu'
+
+steps:
+- task: UseDotNet@2
+  displayName: 'Use .Net Core sdk 2.1.x'
+  inputs:
+    version: 2.1.x
+
+- task: UseDotNet@2
+  displayName: 'Use .Net Core sdk 3.0.x'
+  inputs:
+    version: 3.0.x
+
+- task: NuGetToolInstaller@0
+  displayName: 'Use NuGet 4.9.1'
+  inputs:
+    versionSpec: 4.9.1
+
+- task: NuGetCommand@2
+  displayName: 'NuGet restore'
+  inputs:
+    restoreSolution: '$(System.DefaultWorkingDirectory)\Microsoft.Bot.Builder.sln'
+
+- task: VSBuild@1
+  displayName: 'Build solution Microsoft.Bot.Builder.sln'
+  inputs:
+    solution: Microsoft.Bot.Builder.sln
+    vsVersion: 16.0
+    msbuildArgs: '-p:DefineConstants="FUNCTIONALTESTS"'
```

Comment:
Building different binaries depending on what tests are run should be avoided. Instead of using precompiler directives to select tests, we are using the test attribute [TestCategory("FunctionalTests")] for functional tests. The automated builds running those tests use the argument "--filter TestCategory=FunctionalTests". The builds running other tests use the argument '--filter "TestCategory!=IgnoreInAutomatedBuild&TestCategory!=FunctionalTests"'. You can find examples in PR #3124, release [DotNet-Functional-Tests-Linux-from-signed-yaml](https://fuselabs.visualstudio.com/SDK_v4/_releaseDefinition?definitionId=87&_a=definition-tasks&environmentId=88), and build [BotBuilder-DotNet-master-CI-PR](https://fuselabs.visualstudio.com/SDK_v4/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=457).

Reply:
@matiasroldan6 Can you remove the argument as Bruce suggested? then we can take.